app/vmctl/README.md

```diff
@@ -432,6 +436,79 @@ Found 2 blocks to import. Continue? [Y/n] y
 2020/02/23 15:51:07 Total time: 7.153158218s
 ```
 
+## Migrating data by remote read protocol
+
+`vmctl` supports the `remote-read` mode for migrating data from any database which supports Prometheus remote read protocol 
+to VictoriaMetrics time-series database.
+
+See `./vmctl remote-read --help` for details and full list of flags.
+
+Migration is based on the remote reading and allows transparently fetch samples. 
+Remote read protocol supports two different modes `SAMPLES` and `STREAMED_XOR_CHUNKS`. You can read more about it 
+[here](https://prometheus.io/blog/2019/10/10/remote-read-meets-streaming/).
+
+Please notice at that moment not all TSDB support `STREAMED_XOR_CHUNKS` mode and can handle stream requests.
+
+To start the migration process some flags should be defined:
+1. `--remote-read-src-addr` - address to perform read from;
+2. `--vm-addr` - address for single-node VM is usually equal to `--httpListenAddr`, and for cluster version;
+is equal to `--httpListenAddr` flag of vminsert component (for example `http://<vminsert>:8480/insert/<accountID>/prometheus`);
+3. `--remote-read-filter-time-start` - the time filter in RFC3339 format to select timeseries with timestamp equal or higher than provided value. E.g. '2020-01-01T20:07:00Z';
+4. `--remote-read-step-interval` - split export data into chunks. Valid values are `month, day, hour, minute`;
+
+As soon as required flags are provided and all endpoints are accessible, `vmctl` will start the fetch process 
+from remote read API. Basically, it just fetches all timeseries from the provided database and sends import requests 
+for each timeseries and pass results to VM importer. VM importer then accumulates received samples in batches and 
+sends import requests to VM.
+
+The importing process example for local installation of Prometheus
+and single-node VictoriaMetrics(`http://localhost:8428`):
+
+```
+./vmctl remote-read \
+--remote-read-src-addr=http://127.0.0.1:9091 \
+--remote-read-filter-time-start=2021-10-18T00:00:00Z \
+--remote-read-step-interval=hour \
+--vm-addr=http://127.0.0.1:8428 \
+--vm-concurrency=6
+
+Split defined times into 8798 ranges to import. Continue? [Y/n]
+VM worker 0:↘ 127177 samples/s
+VM worker 1:↘ 140137 samples/s
+VM worker 2:↘ 151606 samples/s
+VM worker 3:↘ 130765 samples/s
+VM worker 4:↘ 131904 samples/s
+VM worker 5:↘ 132693 samples/s
+Processing ranges: 8798 / 8798 [█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████] 100.00%
+2022/10/19 16:45:37 Import finished!
+2022/10/19 16:45:37 VictoriaMetrics importer stats:
+  idle duration: 6m57.793987511s;
```

Comment:
hm, idle duration is 7 times bigger than import duration. It means vmctl was mostly waiting for responses from the source. Do you know why so?

Reply:
This test was in a single Prometheus instance but this instance was a part of Thanos setup I think this is the first reason. Second, I think that system should collect all series for one hour and after that pack them into protobuf representation and send. After we should unpack it and prepare samples.